ipyvolume/transferfunction.py

```diff
@@ -164,3 +165,141 @@ def control(self, max_opacity=0.2):
+    :rtype: ipyvolume TransferFunction
+
+    :Example:
+    >>> import ipyvolume as ipv
+    >>> rgb = (0, 255, 0)  # RGB value for green
+    >>> green_tf = ipv.transfer_function.linear_transfer_function(rgb)
+    >>> ds = ipv.datasets.aquariusA2.fetch()
+    >>> ipv.volshow(ds.data[::4,::4,::4], tf=green_tf)
+    >>> ipv.show()
+
+    .. seealso:: matplotlib_transfer_function()
+    """
+    _num_elements = 256  # length of rgba transfer function array
+    r, g, b = [value/255. for value in rgb_values]  # rescales 0-255 to float
+    opacity = np.linspace(min_opacity, max_opacity, num=_num_elements)
+    if reverse_opacity: opacity = np.flip(opacity, axis=0)
+    rgba = np.transpose(np.stack([[r] * _num_elements,
+                                  [g] * _num_elements,
+                                  [b] * _num_elements,
+                                  opacity]))
+    transfer_function = TransferFunction(rgba=rgba)
+    return transfer_function
+
+
+def matplotlib_transfer_function(colormap_name,
+                                 min_opacity=0,
+                                 max_opacity=0.05,
+                                 reverse_colormap=False,
+                                 reverse_opacity=False):
+    """Transfer function from matplotlib colormaps.
+
+    :param colormap_name: name of matplotlib colormap
+    :param min_opacity: Minimum opacity, default value is 0.
+        Lowest possible value is 0, optional.
+    :param max_opacity: Maximum opacity, default value is 0.05.
+        Highest possible value is 1.0, optional.
+    :param reverse_colormap: reversed matplotlib colormap, optional.
+    :param reverse_opacity: Linearly decrease opacity, optional.
+    :type colormap_name: str
+    :type min_opacity: float, int
+    :type max_opacity: float, int
+    :type reverse_colormap: bool
+    :type reverse_opacity: bool
+    :return: transfer_function
+    :rtype: ipyvolume TransferFunction
+
+    :Example:
+    >>> import ipyvolume as ipv
+    >>> rgb = (0, 255, 0)  # RGB value for green
+    >>> green_tf = ipv.transfer_function.matplotlib_transfer_function('viridis')
+    >>> ds = ipv.datasets.aquariusA2.fetch()
+    >>> ipv.volshow(ds.data[::4,::4,::4], tf=green_tf)
+    >>> ipv.show()
+
+    .. seealso:: linear_transfer_function()
+    """
+    _num_elements = 256  # length of rgba transfer function array
+    cmap = matplotlib.cm.get_cmap(name=colormap_name)
+    rgba = np.array([cmap(i) for i in np.linspace(0, 1, _num_elements)])
+    if reverse_colormap: rgba = np.flip(rgba, axis=0)
+    # Create opacity values to overwrite default matplotlib opacity=1.0
+    opacity = np.linspace(min_opacity, max_opacity, num=_num_elements)
+    if reverse_opacity: opacity = np.flip(opacity, axis=0)
+    rgba[:,-1] = opacity # replace opacity=1 with actual opacity
+    transfer_function = TransferFunction(rgba=rgba)
+    return transfer_function
+
+
+def load_transfer_functions(include_rgb_linear=True,
```

Comment:
I would not call it load (get?), and I'd make it 'singleton' like, so it will always return the same widgets, what do you think? (it will basically cache the widgets).
My guess is this will be used as part of a gui to select a particular colormap?

Reply:
Let's call it 'predefined_transfer_functions'.

I'm not sure I understand correctly: do you want this to stay a function but always return the same dictionary, or have me make a singleton class for it?

I have no major plans for a colormap gui. But I do want to set future efforts at a GUI up for success as much as possible, so if you have advice on things I'm doing (or not doing here I'm happy to hear it.

What I want in the short term is a bunch of transfer function widgets so I can easily switch between them programmatically for the multivolume rendering I'm working on now.

Reply:
What I would like to see, is when a transfer function is created with the same set of parameters (same key for the dictionary), that it returns the same transfer function. That will limit the number of widgets created, what do you think?

Reply:
I'm not sure how best to implement the idea, but I think it's a good strategy/goal.

Reply:
sth like the:
```
tf_cache = {}
def get_transfer_functions(...)
  tf = tf_cache.get(key)
  if not tf:
    tf = ....
    tf_cache[key] = tf
  ...
```